clang/lib/Driver/Driver.cpp

```diff
@@ -4939,6 +4957,14 @@ class OffloadingActionBuilder final {
             auto &TargetInfo = std::get<1>(TargetActionInfo);
             A = C.getDriver().ConstructPhaseAction(C, Args, CurPhase, A,
                                                    AssociatedOffloadKind);
+            // No need to generate the integration header if
+            // -fsycl-no-use-integration-header is specified.
+            Arg *UIHArg =
+                Args.getLastArg(options::OPT_fsycl_use_integration_headers,
+                                options::OPT_fno_sycl_use_integration_headers);
+            if (UIHArg && UIHArg->getOption().matches(
+                              options::OPT_fno_sycl_use_integration_headers))
```

Comment:
This check is used pretty often and doesn't seem to really need to check 'last one' to see if integration headers have been disabled.  Maybe replace with !hasFlag(use, no_use, true)?

Reply:
I am a bit confused about the behavior of hasFlag:
if the user says: '-no_use -use' on the command-line, will it still work correctly?

Reply:
It should, hasFlag does the 'last one wins' logic for you.

-no-use -use -> true
-use -no-use -> false

Reply:
Thanks @mdtoguchi.  Done in all places except one, where the other form is needed.